ond.py

```diff
@@ -90,27 +90,28 @@ def run_protocol(self, config_):
                         "FeatureExtraction", file_list
                     )
 
-                    results = {}
-                    results["detection"] = algorithm.execute(
-                        "WorldDetection", features_dict, logits_dict,
-                        red_light, round_id=round_id
-                    )
-                    results["classification"] = algorithm.execute(
-                        "NoveltyClassification", features_dict, logits_dict,
-                        round_id=round_id
-                    )
+                    if not config["feature_extraction_only"]:
```

Comment:
I would recommend adding `continue` after checking that `feature_extraction_only` config option. Although it would require that same condition in characterization.

Reply:
I'm intentionally trying to avoid using `continue` because I feel it makes it more difficult to follow the control flow logic of the loop (though, obviously, I'm open to using it if there's a compelling reason).

Could you clarify why the same condition would be necessary for characterization?

Reply:
If we use `continue` then it wouldn't execute anything in a round. However since characterization occurs after all round we would still need to skip characterization. Based on the quick glance `not config[`feature_extraction_only`] felt a little difficult to follow. Especially for characterization where the first time I saw it I was like is that in the loop or out of it.

Reply:
@as6520 Updated to `continue` if `config["feature_extraction_only"]` is `True`. This also addresses the save features comment.